components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -130,8 +130,29 @@ public OAuth2ClientValidationResponseDTO validateClientInfo(String clientId, Str
         }
 
         try {
-            OAuthAppDAO oAuthAppDAO = new OAuthAppDAO();
-            OAuthAppDO appDO = oAuthAppDAO.getAppInformation(clientId);
+            if (StringUtils.isBlank(clientId)) {
+                throw new InvalidOAuthClientException("Invalid client_id. No OAuth application has been registered " +
```

Comment:
Is the error msg is correct?

Reply:
Since the end-user is redirected to the error page, having a more specific error message such as client_id is null or empty is not necessary.